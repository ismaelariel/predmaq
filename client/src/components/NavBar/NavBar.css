.div_navbar_container {
    width: 100%;
    height: 8%;
    display: flex;
    position: relative;
    background-color: #0000ff;
}

.div_navbar_content {
    flex: 1;
    width: auto;
    height: 100%;
    margin: 0 8px;
    color: #ffffff;
    position: relative;
}

.div_navbar_content_app_name {
    display: flex;
    align-items: center;
    justify-content: start;
}

.div_navbar_content_datetime {
    display: flex;
    align-items: center;
    justify-content: end;
    font-size: x-large;
    font-weight: bold;
}

.text_datatime_navbar {
    margin: 0;
    padding: 0;
}