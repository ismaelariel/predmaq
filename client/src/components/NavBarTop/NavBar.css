.div_navbar_container {
    width: 100%;
    height: 8%;
    display: flex;
    position: relative;
    background-color: #0000ff;
}

.div_navbar_content {
    flex: 1;
    width: auto;
    height: 100%;
    margin: 0 8px;
    display: flex;
    color: #ffffff;
    position: relative;
    align-items: center;
}