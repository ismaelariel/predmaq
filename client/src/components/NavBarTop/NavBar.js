import React, {useState, useEffect} from "react";

import "./NavBar.css";

const NavBar = () => {
    const [dateBase, setDateBase] = useState("");

    useEffect(() => {
        const date = new Date();
        
        const stateTimeout = setTimeout(() => {
            setDateBase(date.toLocaleString());
        }, 1000);

        return () => {clearTimeout(stateTimeout)};

    }, [dateBase]);

    return(
        <div className="div_navbar_container">
            <div className="div_navbar_content">
                <h4>PredMaq</h4>
            </div>
            <div className="div_navbar_content">
                <small>{dateBase}</small>
            </div>
        </div>
    );
};

export default NavBar;