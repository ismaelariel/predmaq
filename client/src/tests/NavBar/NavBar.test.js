import React from "react";
import renderer from "react-test-renderer";
import NavBar from "../../App";

const mockTimeOutAsync = (like) => {
    setTimeout(() => {
        like(true);
    }, 1000);
}

jest.useFakeTimers();
jest.spyOn(global, 'setTimeout');

describe('NavBar Component View', () => {
    test('test mock datetime callbacks navbar', async () => {
        const timeCallback = (data) => {
            expect(data).toBeTruthy();
        }

        mockTimeOutAsync(timeCallback);

        jest.runAllTimers();

        expect(setTimeout).toHaveBeenCalledTimes(1);
        expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);
    });

    it("render navbar view", async () => {
        const component = renderer.create(
            <NavBar />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
