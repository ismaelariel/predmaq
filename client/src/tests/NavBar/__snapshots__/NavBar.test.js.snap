// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component View render navbar view 1`] = `
<div
  className="div_navbar_container"
>
  <div
    className="div_navbar_content div_navbar_content_app_name"
  >
    <h4>
      PredMaq
    </h4>
  </div>
  <div
    className="div_navbar_content div_navbar_content_datetime"
  >
    <span
      className="text_datatime_navbar"
    />
  </div>
</div>
`;
